src/miral/display_configuration.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright © 2018 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: Alan Griffiths <alan@octopull.co.uk>
+ */
+
+#include "miral/display_configuration.h"
+#include "static_display_config.h"
+#include "shared_data.h"
+
+#include <mir/server.h>
+
+#include <fstream>
+#include <sstream>
+
+struct miral::DisplayConfiguration::Self : StaticDisplayConfig {};
+
+miral::DisplayConfiguration::DisplayConfiguration() :
+    self{std::make_shared<Self>()}
+{
+}
+
+void miral::DisplayConfiguration::select_layout(std::string const& layout)
+{
+    self->select_layout(layout);
+}
+
+void miral::DisplayConfiguration::operator()(mir::Server& server) const
+{
+    std::string const name = rootname + ".display";
+    std::string config_roots;
+
+    if (auto config_home = getenv("XDG_CONFIG_HOME"))
+        (config_roots = config_home) += ":";
```

Comment:
This is an interesting line of code. Any particular reason its not simply `config_roots = config_home + ":";`?